hack/prom_metric_linter.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+#
+# Copyright 2022 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+linter_image_tag="v0.0.1"
+
+PROJECT_ROOT="$(readlink -e "$(dirname "${BASH_SOURCE[0]}")"/../)"
+export METRICS_COLLECTOR_PATH="${METRICS_COLLECTOR_PATH:-${PROJECT_ROOT}/tools/prom-metrics-collector}"
+
+if [[ ! -d "$METRICS_COLLECTOR_PATH" ]]; then
+    echo "Invalid METRICS_COLLECTOR_PATH: $METRICS_COLLECTOR_PATH is not a valid directory path"
+    exit 1
+fi
+
+# Parse command-line arguments
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+    --operator-name=*)
+        operator_name="${1#*=}"
+        shift
+        ;;
+    --sub-operator-name=*)
+        sub_operator_name="${1#*=}"
+        shift
+        ;;
+    *)
+        echo "Invalid argument: $1"
+        exit 1
+        ;;
+    esac
+done
+
+# Get the metrics list
+go build -o _out/prom-metrics-collector "$METRICS_COLLECTOR_PATH/..."
+json_output=$(_out/prom-metrics-collector 2>/dev/null)
+
+# Select container runtime
+source "${PROJECT_ROOT}"/hack/config.sh
+
+# Check if runs as part of Openshift-ci. If so, run directly
+if [[ "${OPENSHIFT_CI}" == "true" ]]; then
+  errors=$(/usr/bin/prom-metrics-linter \
+      --metric-families="$json_output" \
+      --operator-name="$operator_name" \
+      --sub-operator-name="$sub_operator_name" 2>/dev/null)
+# Else, run in a container, using the linter image
+else
+  errors=$($KUBEVIRT_CRI run -i "quay.io/kubevirt/prom-metrics-linter:$linter_image_tag" \
+      --metric-families="$json_output" \
+      --operator-name="$operator_name" \
+      --sub-operator-name="$sub_operator_name" 2>/dev/null)
+fi
+
+# Check if there were any errors, if yes print and fail
+if [[ $errors != "" ]]; then
```

Comment:
> If the linter fail as expected and pass as expected I'm OK with removing the use of `$errors`.

ACK thanks

> Does `implemented in the same format` mean this script is copied into multiple repos with the same contents? If so I would prefer to store it in one common place instead.

I don't think we can do that for the script but definitely should be the collector and linter (as you are already so kudos).

Reply:
The linter doesn't produce an error when metrics are failing. This is why we need it.

Reply:
I'd recommend fixing the behaviour of the linter then, returning `>0` on failure should be a given for the tool so you don't need to rely on output like this.

Reply:
@avlitman Is it possible to change the linter's behavior, so it will fail when metrics are failing?
@lyarwood Anyway, I think this should be a followup, since the linter was added to 5 different repos as is.

Reply:
> Does `implemented in the same format` mean this script is copied into multiple repos with the same contents? If so I would prefer to store it in one common place instead.

No, we already moved most of the code to general place, you can take a look a the pr's.
we did are best and already been through many changes request and the linter is in the best way we could at the moment. 

This pr add the linter not fix the linter, so please if you want to fix the linter, we will open an issue to /monitoring.

It's important to us to to add this linter asap to make sure no wrong metrics names is added since customer use metrics in external tools and it's not an easy removal for the old names.

I don't see a reason for the linter to fail in the linter itself- it's a tool to get error with metrics names and the user should be able to decide if to fail the ci on it or not.

We can create a follow-up with all the notes to the linter, but for now this is the current state.